LanguageFeatures/Extension-types/exhaustiveness_object_A01_t03.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with a sealed class as a
+/// matched type are always exhaustive
+///
+/// @description Check that it is no compile-time error if the matched value
+/// type of a switch expression is a sealed class as and the set of cases is an
+/// exhaustive set of object patterns with extension types
```

Comment:
Yes, this one should work today!

(We _do_ have a decision that object patterns whose type is an extension type should be considered to have the erasure as its type during exhaustiveness computations).